test/jdk/java/nio/channels/FileChannel/TransferToChannel.java

```diff
@@ -22,25 +22,44 @@
  */
 
 /* @test
- * @bug 4652496
+ * @bug 4652496 8274112
  * @summary Test transferTo with different target channels
+ * @library /test/lib
+ * @build jdk.test.lib.RandomFactory
  * @run main TransferToChannel
  * @run main/othervm -Djdk.nio.enableFastFileTransfer TransferToChannel
+ * @key randomness
  */
 
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
 import java.nio.channels.FileChannel;
 import java.nio.channels.WritableByteChannel;
-import java.nio.ByteBuffer;
-import java.io.*;
+import java.nio.file.Files;
+import java.util.Arrays;
 import java.util.Random;
+import jdk.test.lib.RandomFactory;
 
 public class TransferToChannel {
 
-    static File file;
-    static File outFile;
-    static FileChannel in;
-    // Chunk size should be larger than FileChannelImpl.TRANSFER_SIZE for good test
-    static int CHUNK_SIZE = 1024 * 9;
+    private static final Random RAND = RandomFactory.getRandom();
+
+    private static final int FILE_SIZE = 1000*1024;
```

Comment:
Could use a comment regarding the reason the file size is what it is

Reply:
I don't recall why that particular value was chosen except in relation to other constants. Will add some comments.

Reply:
Fixed by commit c72b4b58f57d998423fb94b17ac950185109b933.